homeassistant/components/sensor/metoffice.py

```diff
@@ -0,0 +1,159 @@
+    ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['datapoint==0.4.3']
+
+CONF_ATTRIBUTION = "Data provided by the Met Office"
+CONF_MO_API_KEY = 'api_key'
+
+SCAN_INTERVAL = timedelta(minutes=35)
+LAST_UPDATE = 0
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'name': ['Station Name', None],
+    'weather': ['Weather', None],
+    'temperature': ['Temperature', TEMP_CELSIUS],
+    'feels_like_temperature': ['Feels Like Temperature', TEMP_CELSIUS],
+    'wind_speed': ['Wind Speed', 'mps'],
+    'wind_direction': ['Wind Direction', None],
+    'wind_gust': ['Wind Gust', 'mps'],
+    'visibility': ['Visibility', 'km'],
+    'uv': ['UV', None],
+    'precipitation': ['Probability of Precipitation', '%'],
+    'humidity': ['Humidity', '%']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Required(CONF_MO_API_KEY): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    import datapoint as dp
+    datapoint = dp.connection(api_key=config.get(CONF_MO_API_KEY))
+
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    try:
+        site = datapoint.get_nearest_site(longitude=hass.config.longitude,
+                                          latitude=hass.config.latitude)
+    except dp.exceptions.APIException as err:
+        _LOGGER.error("Received error from Met Office Datapoint: %s", err)
+        return False
+
+    if not site:
+        _LOGGER.error("Unable to get nearest Met Office forecast site")
+        return False
+    else:
+
+        # Get data
+        data = MetOfficeCurrentData(hass, datapoint, site)
+        try:
+            data.update()
+        except (ValueError, dp.exceptions.APIException) as err:
+            _LOGGER.error("Received error from Met Office Datapoint: %s", err)
+            return False
+
+        # Add
+        add_devices([MetOfficeCurrentSensor(site, data, variable)
+                     for variable in config[CONF_MONITORED_CONDITIONS]])
+        return True
+
+
+class MetOfficeCurrentSensor(Entity):
+    """Implementation of a Met Office current sensor."""
+
+    def __init__(self, site, data, condition):
+        """Initialize the sensor."""
+        self.site = site
+        self.data = data
+        self._condition = condition
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'Met Office {}'.format(SENSOR_TYPES[self._condition][0])
```

Comment:
This is in keeping with the bom component. Also you comment here is at odds with your comment on the weather component about using `CONF_NAME`. 

Do you suggest using `CONF_NAME` if it has been set and `DEFAULT_NAME` otherwise?